From c5328b7d87942abd73d3a904df0554b41d4dc1f0 Mon Sep 17 00:00:00 2001
From: Christian Hewitt <christianshewitt@gmail.com>
Date: Wed, 27 Jul 2022 13:39:23 +0000
Subject: [PATCH 81/92] WIP: arm64: dts: meson: add audio lineout to WeTek
 Play2

Add support for the ES7134LV headphone output on the WeTek Play2

Signed-off-by: Christian Hewitt <christianshewitt@gmail.com>
---
 .../boot/dts/amlogic/meson-gxbb-wetek-play2.dts   | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxbb-wetek-play2.dts b/arch/arm64/boot/dts/amlogic/meson-gxbb-wetek-play2.dts
index 505ffcd8eb76..e206d658d157 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxbb-wetek-play2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxbb-wetek-play2.dts
@@ -21,6 +21,13 @@ spdif_dit: audio-codec-0 {
 		sound-name-prefix = "DIT";
 	};
 
+	i2s_codec: audio-codec-1 {
+		#sound-dai-cells = <0>;
+		compatible = "everest,es7134";
+		VDD-supply = <&vcc_5v>;
+		status = "okay";
+	};
+
 	leds {
 		led-wifi {
 			label = "wetek-play:wifi-status";
@@ -74,6 +81,10 @@ dai-link-2 {
 			codec-0 {
 				sound-dai = <&aiu AIU_HDMI CTRL_I2S>;
 			};
+
+			codec-1 {
+				sound-dai = <&i2s_codec>;
+			};
 		};
 
 		dai-link-3 {
@@ -96,7 +107,9 @@ codec-0 {
 
 &aiu {
 	status = "okay";
-	pinctrl-0 = <&spdif_out_y_pins>;
+	pinctrl-0 = <&i2s_am_clk_pins>, <&i2s_out_ao_clk_pins>,
+		    <&i2s_out_lr_clk_pins>, <&i2s_out_ch01_ao_pins>,
+		    <&spdif_out_y_pins>;
 	pinctrl-names = "default";
 };
 
-- 
2.34.1

